src/main/java/io/openshift/booster/service/impl/JdbcProductStore.java

```diff
@@ -37,17 +37,9 @@ public JdbcProductStore(JDBCClient db) {
 
   @Override
   public Single<JsonObject> create(JsonObject item) {
-    if (item == null) {
-      return Single.error(new IllegalArgumentException("The item must not be null"));
-    }
-    if (item.getString("name") == null || item.getString("name").isEmpty()) {
-      return Single.error(new IllegalArgumentException("The name must not be null or empty"));
-    }
-    if (item.getInteger("stock", 0) < 0) {
-      return Single.error(new IllegalArgumentException("The stock must greater or equal to 0"));
-    }
-    if (item.containsKey("id")) {
-      return Single.error(new IllegalArgumentException("The created item already contains an 'id'"));
+    Exception error = validateRequestBody(item);
```

Comment:
I would avoid the null check and return an Optional.

Reply:
It is the good point. I'll do it. Thanks